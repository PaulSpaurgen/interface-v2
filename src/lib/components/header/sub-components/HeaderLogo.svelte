<a href="/">
	<img src="/_oysterstatic/logo/logo-name.svg" alt="Marlin Logo" width="140" />
</a>
