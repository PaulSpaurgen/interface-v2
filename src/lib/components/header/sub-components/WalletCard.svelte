<script lang="ts">
	import Text from '$lib/atoms/texts/Text.svelte';

	export let imageSrc: string;
	export let title: string;

	const styles = {
		card: 'bg-base-200 rounded-lg overflow-hidden p-4 w-1/2 flex flex-col items-center'
	};
</script>

<div class={styles.card}>
	{#if !!imageSrc}
		<img src={imageSrc} alt={title} width="64px" />
	{/if}
	<Text text={title} variant="h4" styleClass="mt-4 mb-6 font-semibold font-sans" />
	<slot name="button" />
</div>
