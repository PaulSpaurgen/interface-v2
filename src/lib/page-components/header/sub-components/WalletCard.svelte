<script lang="ts">
	import Text from '$lib/atoms/texts/Text.svelte';

	export let imageSrc: string;
	export let title: string;

	const styles = {
		card: 'bg-base-200 rounded-lg overflow-hidden p-4 w-1/2 flex flex-col items-center'
	};
</script>

<div class={styles.card}>
	{#if !!imageSrc}
		<img src={imageSrc} alt={title} width="64px" />
	{/if}
	<Text text={title} styleClass="mt-4 mb-6 text-[19px] font-semibold font-sans" />
	<slot name="button" />
</div>
