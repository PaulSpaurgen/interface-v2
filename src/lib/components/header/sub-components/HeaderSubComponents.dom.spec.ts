
import { render } from '@testing-library/svelte';
import WalletCard from './WalletCard.svelte';
import HeaderLogo from './HeaderLogo.svelte';
import { describe, it, expect } from 'vitest';
import html from 'svelte-htm';

describe('WalletCard', () => {
    test('renders the WalletCard component properly', () => {
        const { container } = render(WalletCard);
        expect(container).toMatchSnapshot();
    });

    it('renders the title text', () => {
        const titleText = 'Sample Title';

        const { getByText } = render(WalletCard, {
            props: {
                title: titleText,
                imageSrc: '', // Pass an empty string because we are testing the title here
            },
        });
        expect(getByText(titleText)).toBeTruthy();
    });

    it('renders an image when imageSrc is provided', () => {
        const testImageUrl = 'test-image.jpg';
        const testTitle = 'Test Image';

        const { getByAltText } = render(WalletCard, {
            props: {
                imageSrc: testImageUrl,
                title: testTitle,
            },
        });

        const image = getByAltText(testTitle);
        expect(image).toBeTruthy();
        expect(image.getAttribute('src')).toBe(testImageUrl);
    });

    it('does not render an image when imageSrc is not provided', () => {
        const { container } = render(WalletCard, {
            props: {
                title: 'No Image Title',
                imageSrc: ''
            },
        });

        // Here we are checking that no <img> elements are present in the container.
        expect(container.querySelector('img')).toBeNull();
    });


    it("allows content to be added via slots", async () => {
        const slotContentText = 'Click me';
        const { getByText } = render(html`
                <${WalletCard} title=${'Card with Button'} imageSrc='example.png'>
                 <div slot='button'><button>${slotContentText}</button></div>
                </${WalletCard}>
              `);
        expect(getByText(slotContentText)).toBeTruthy();
    });
});

describe('HeaderLogo', () => {
    test('renders the HeaderLogo component properly', () => {
        const { container } = render(HeaderLogo);
        expect(container).toMatchSnapshot();
    });
    it('renders both images with correct sources and one is initially hidden', () => {
        const { getAllByAltText } = render(HeaderLogo);

        const images = getAllByAltText('Marlin Logo');

        // Expect two images to be found
        expect(images.length).toBe(2);

        // Check if the right logos have the right src
        const largeScreenLogo = images.find((img) => img.getAttribute('class')?.includes('sm:block'));
        const smallScreenLogo = images.find((img) => img.getAttribute('class')?.includes('sm:hidden'));

        // Since we cannot test screen sizes here, we'll just check if the src is correct
        expect(largeScreenLogo?.getAttribute('src')).toBe('/logo/logo-name.svg');
        expect(smallScreenLogo?.getAttribute('src')).toBe('/logo/marlin-logo.svg');
    });
});