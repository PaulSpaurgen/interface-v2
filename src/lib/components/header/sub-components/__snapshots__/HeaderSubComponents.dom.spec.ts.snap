// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConnectWalletButton > renders the ConnectWalletButton component properly 1`] = `
<body>
  <onboard-v2
    style="all: initial;"
  />
  <style>
    
    
@font-face {
  font-family: 'Inter';
  font-style:  normal;
  font-weight: 300 600;
  font-display: swap;
  src: url("https://rsms.me/inter/font-files/Inter-roman.var.woff2?v=3.19") format("woff2");
}

  
  </style>
  <div>
    <a
      href="/"
    >
      <img
        alt="Marlin Logo"
        class="hidden sm:block"
        src="/logo/logo-name.svg"
        width="140"
      />
       
      <img
        alt="Marlin Logo"
        class="block sm:hidden"
        src="/logo/marlin-logo.svg"
        width="40"
      />
    </a>
    <!--&lt;HeaderLogo&gt;-->
  </div>
</body>
`;

exports[`HeaderLogo > renders the HeaderLogo component properly 1`] = `
<body>
  <onboard-v2
    style="all: initial;"
  />
  <style>
    
    
@font-face {
  font-family: 'Inter';
  font-style:  normal;
  font-weight: 300 600;
  font-display: swap;
  src: url("https://rsms.me/inter/font-files/Inter-roman.var.woff2?v=3.19") format("woff2");
}

  
  </style>
  <div>
    <a
      href="/"
    >
      <img
        alt="Marlin Logo"
        class="hidden sm:block"
        src="/logo/logo-name.svg"
        width="140"
      />
       
      <img
        alt="Marlin Logo"
        class="block sm:hidden"
        src="/logo/marlin-logo.svg"
        width="40"
      />
    </a>
    <!--&lt;HeaderLogo&gt;-->
  </div>
</body>
`;

exports[`WalletCard > renders the WalletCard component properly 1`] = `
<body>
  <onboard-v2
    style="all: initial;"
  />
  <style>
    
    
@font-face {
  font-family: 'Inter';
  font-style:  normal;
  font-weight: 300 600;
  font-display: swap;
  src: url("https://rsms.me/inter/font-files/Inter-roman.var.woff2?v=3.19") format("woff2");
}

  
  </style>
  <div>
    <div
      class="flex w-1/2 flex-col items-center overflow-hidden rounded-lg bg-base-200 p-4"
    >
      <img
        alt="small card"
        src="example.png"
        width="64px"
      />
       
      <div
        class="mt-4 mb-6 font-semibold font-sans text-xl font-bold"
        data-testid="text"
      >
        small card
      </div>
      <!--&lt;Text&gt;-->
       
    </div>
    <!--&lt;WalletCard&gt;-->
  </div>
</body>
`;
